interceptors/interceptors.go

```diff
@@ -198,16 +201,22 @@ func HystrixClientInterceptor() grpc.UnaryClientInterceptor {
 		}
 		newCtx, cancel := context.WithCancel(ctx)
 		defer cancel()
-		return hystrix.Do(options.hystrixName, func() (err error) {
+		var err error
+		_ = hystrix.Do(options.hystrixName, func() (e error) {
 			defer func() {
 				if r := recover(); r != nil {
 					err = errors.Wrap(fmt.Errorf("Panic inside hystrix Method: %s, req: %v, reply: %v", method, req, reply), "Hystrix")
 					log.Error(ctx, "panic", r, "method", method, "req", req, "reply", reply)
 				}
 			}()
 			defer notifier.NotifyOnPanic(newCtx, method)
-			return invoker(newCtx, method, req, reply, cc, opts...)
+			// error assigns back to the err object out of hystrix anyway
+			err = invoker(newCtx, method, req, reply, cc, opts...)
+			// error could be suppressed in hystrix wrapper by the function from options
+			return options.hystrixErrorSuppressor(err)
```

Comment:
Instead of treating `hystrixErrorSuppressor` as always present and relying on the function to be implicitly initialized as a passthrough, let's only call `hystrixErrorSuppressor` if non-nil. Basically if `hystrixErrorSuppressor` exists, then it was provided by the caller.

```
if options.hystrixErrorSupressor != nil {
  return options.hystrixErrorSupressor(invoker(newCtx, method, req, reply, cc, opts...))
} else {
  return invoker(newCtx, method, req, reply, cc, opts...)
}
```

Reply:
yea, I was doing this way and I thought defining a default Suppressor could make the code cleaner. However, I think checking non-nil is better for readability. I will make the change.

Reply:
changed and added one unit test https://github.com/carousell/Orion/pull/155/commits/282124e021c025599fa01e4c519a76fd460a779e#diff-c52dbca06ba660883cf2e48f7e11d68dcab4d17b453d068837aff683f3de61cbR1